src/evict/evict_page.c

```diff
@@ -731,6 +736,16 @@ __evict_review_obsolete_time_window(WT_SESSION_IMPL *session, WT_REF *ref)
     if (__wt_page_is_modified(ref->page))
         return (0);
 
+    /* Limit the number of btrees that can be cleaned up. */
+    if (btree->obsolete_tw_pages == 0 &&
+      conn->heuristic_controls.obsolete_tw_btree_count >=
+        conn->heuristic_controls.obsolete_tw_btree_max)
+        return (0);
+
+    /* Don't add more cache pressure. */
+    if (__wt_eviction_dirty_needed(session, NULL) || __wt_eviction_updates_needed(session, NULL))
```

Comment:
@kommiharibabu, should I add the same checks as in checkpoint cleanup?

Reply:
No. As of now, the checkpoint cleanup leads to more clean pages and it is already taken care of. When we are changing it to perform obsolete page dirty, adding the dirty check make sense.

Reply:
I have refined the checks in 988de07da96ff6aad650c8cc64b94854b36abff6